src/env.c

```diff
@@ -1,6 +1,26 @@
 #include "rlang.h"
 
+
+bool r_is_unbound_value(SEXP x) {
+  return x == R_UnboundValue;
+}
+
 SEXP r_mut_env_parent(SEXP env, SEXP new_parent) {
```

Comment:
I like the `void` return value, but taking a `SEXP*` would imply that the pointer holding the SEXPREC would change, which won't be the case.

I'd like to typedef the `SEXPREC` to `sexp`, so we'd use `sexp*` instead of `SEXP` to make semantics more explicit.